docs/design/configuration.md

```diff
@@ -0,0 +1,141 @@
+# Kubernetes Component Configuration
+
+## Abstract
+
+A generic ``Configuration`` object to store command-line flags that is used for
+configuration of ``Kubernetes`` components living on top of ``etcd`` is proposed
+in this proposal.
+
+The main focus of this proposal to solve dynamic configuration problem of components
+and encapsulate configuration information to increase overall system quality and to
+create a flexible configuration model for ``Kubernetes`` components.
+
+``Kubernetes`` component configuration happens on the command line, and this proposal
+builds on that fundamental principle to be coherent and easy for admins to comprehend.
+
+## Problem
+
+Currently command-line flags are used for component configurations in ``Kubernetes``.
+But, this approach is limiting in a distributed computing environment. Specifically:
+
+* Hard to synchronize across replicated components.
+* Hard to change dynamically as needed by continuous services.
+* Hard to change in different versions that may support different configuration options.
+
+## Solution
+
+A new ``Configuration`` resource is proposed to solve the aforementioned problems.
```

Comment:
@bgrant0607 I see. Using `ConfigData` or `ConfigParameters` might be the best choice then. But, since files are also planned to be in it, `ConfigParameters` might not be an option either. In any case, I'll continue with `Config` and we can change the name when the name decision is finalized.